ui/src/forms/core/BaseForm/BaseForm.js

```diff
@@ -24,15 +23,13 @@ export class BaseForm extends Component {
         ? await this.props.editApiMethod(this.props.pid, submitValues)
         : await this.props.createApiMethod(submitValues);
 
-      setTimeout(() => {
-        this.props.sendSuccessNotification(
-          'Success!',
-          this.props.successSubmitMessage
-        );
-        if (this.props.successCallback) {
-          this.props.successCallback(response, submitButton);
-        }
-      }, ES_DELAY);
+      this.props.sendSuccessNotification(
+        'Success!',
```

Comment:
nit: I like this refactoring here and I would say that even the "Success" argument should be handled from the function `sendSuccessNotification`. It was like that before as it was required from the action for the notification part. But I think is clearer if the function just accepts a function. Also, in that case I would say the `successSubmitMessage` is redundant.

Reply:
I agree. Honestly I started to do it, but then I saw it was touching all forms, so better to do it in another PR if it is ok with you.